pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -1074,8 +1076,10 @@ func handleModify(ctxArg interface{}, key string,
 		return
 	}
 
-	if config.PurgeCmd.Counter != oldConfig.PurgeCmd.Counter ||
-		config.LocalPurgeCmd.Counter != oldConfig.LocalPurgeCmd.Counter {
+	// We need to perform a strict check for greater than because the counter might decrease during a snapshot rollback
+	// and the application of the old config.
```

Comment:
Is this because the oldConfig changes (to an older oldConfig with a older and different counter value)?

If so, do we need to worry about this integer wrapping around to zero?
The check for inequality did not have to worry about that.
One way to do that is, when we rollback the config, to update the *Cmd.Counter to the values in the current config.

Reply:
Yep, that's because we apply an oldConfig during the rolling back now.
Could you please explain the specific issue with the integer wrapping around zero?

I can try to test the restoration of the current values of the counters, but it looks dangerous to me. We report the configuration version to the controller when we perform a rollback. My assumption is that the controller uses this specific version as a foundation for future configurations it sends. If that's the case, the restored values may exceed the values in the base version. Consequently, if we simply compare them for inequality, it might trigger another purge or restart. @xyuria-zededa, could you share your opinion on this matter?

Reply:
I assume this Counter is a uint32 (to lazy to look up the actual size). Suppose an app instance has been restarted a lot, to be specific more than 4 billion times. 
Then at some point in time the Counter will go from 2^^32-1 to zero, and the comparison for greater will be incorrect.

I places where one has to deal with wraparound (such as TCP sequence numbers) the comparison is done in a way where it treats the integers as being on a circle instead of a line, and the compaison looks at the distance between the two numbers on the circle to determine which one is before the other.

I'm not saying that users will do 4 billion restarts, but I observe that the old comparison avoided this issue.

Good point about the controller potentially restarting these counters based on the old config, but we should check what it actually does when you restart after a rollback.

Reply:
But it makes sense adding a comment here, plus testing what the controller does when issuing 
1. an updated config (e.g., new ACL) after the rollback - this should not result in a restart of the app instance
2. a restart command after the rollback - this should result in a restart

Reply:
I've tested this scenario with the checks as they used to be before and with the counters' restoration during the rollback process. It looks like it works fine. So, I'll remove this commit and fix the rollback commit correspondingly.
I will push these changes after at least 1 Eden test passes.